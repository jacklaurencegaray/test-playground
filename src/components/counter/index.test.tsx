import React from "react";
import { render, fireEvent, waitForElement } from "@testing-library/react";
import Counter from "./index";
import "@testing-library/jest-dom/extend-expect";

test("Counter: increment works", async () => {
  const { getByText, getByTestId } = render(<Counter />);

  fireEvent.click(getByText("Increment"));

  await waitForElement(() => getByTestId("counter-label"));

  expect(await getByTestId("counter-label")).toHaveTextContent;
});
