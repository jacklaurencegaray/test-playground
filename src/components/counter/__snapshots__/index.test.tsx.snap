// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter: increment works: <p>You have clicked 5000 times.</p> 1`] = `
<p
  data-testid="counter-label"
>
  You have clicked 
  1
   times.
</p>
`;
